lib/streamlit/string_util.py

```diff
@@ -111,3 +112,45 @@ def generate_download_filename_from_title(title_string: str) -> str:
     file_name_string = clean_filename(title_string)
     title_string = snake_case_to_camel_case(file_name_string)
     return append_date_time_to_string(title_string)
+
+
+def is_emoji_valid(emoji: str) -> str:
```

Comment:
A few things about this function:

* The name `is_emoji_valid` makes it sound like it returns a boolean, so I think changing the name to `validate_emoji` would be more appropriate.
* As mentioned in another comment, I think we can move the `None`-check currently done in the `check_emoji` so that this function's signature becomes `def validate_emoji(emoji: Optional[str]) -> str`
* We should add a few unit tests for this function

Reply:
Addressed bullet points 1 and 2 in [4390be7](https://github.com/streamlit/streamlit/pull/4964/commits/4390be7616245c5122e9f0c78172c47bdf268aeb). Should we wait until you can take a look at the regex pattern to create the unit tests for this function? Also given the fact @LukasMasuch is also working on an alternative approach for this validation [here](https://github.com/streamlit/streamlit/pull/5076)

Reply:
Ah, yeah I think it'd be best to hold off on unit tests for now since there are a few potential ways of resolving this, so the tests to write will change depending on what we decide to do.

I'm currently leaning toward Lukas' solution as it seems to get us fully correct emoji support (until new emoji are added, at which point we'll have small amounts of work to do to update our list to be complete) without needing a new dependency to be installed, but I'll ask the rest of the open source team and give people a day or two to weigh in before proceeding with any solution.

Reply:
Alrighty! So this is now using @LukasMasuch proposed approach for emoji validation. LMK what you think!